### PR TITLE
Implement `LineIntervalChart` component

### PR DESCRIPTION
**Pull request by pnavarrc**
Implements the `LineThresholdChart`, the non-metric aware version of `MetricLineThresholdChart`. While implementing #341, I realized that the dependency between metrics and the coloured line is fairly weak, and that we can use the non-metric aware version for both `MetricLineThresholdChart` and for the `SeriesChart`.

I also implemented the `createTimeseriesScales` function to simplify the stories that need scales based on `timeseries.

### CODE REVIEW

File: packages/ui-components/src/components/LineChart/LineChart.stories.tsx [hunk 1]
```diff
@@ -11,39 +12,26 @@ export default {
...
-const xScale = scaleTime({
-  domain: [minDate, maxDate],
-  range: [padding, width - padding],
-});
-
-const yScale = scaleLinear({
-  domain: [minValue, maxValue],
-  range: [height - 2 * padding, padding],
-});
+const { xScale, yScale } = createTimeseriesScales(timeseries, width, height);
```
Comment (mikelehen) [inline packages/ui-components/src/components/LineChart/LineChart.stories.tsx:22]: Nice!

File: packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+export interface LineThresholdChartProps {
+  /**
+   * Array defining the intervals on the y-axis on which the line will
+   * be rendered with the color specified in the interval.
+   */
+  intervals: LineThresholdInterval[];
+  /** Timeseries to render as a line */
+  timeseries: Timeseries<number>;
+  /** d3-scale to transform point dates to pixel positions on the x-axis */
```
Comment (mikelehen) [inline packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx]: Should this be y-axis?

File: packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  yScale: ScaleLinear<number, number>;
+  /** d3-scale to transform point values to pixel positions on the y-axis */
```
Comment (mikelehen) [inline packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx]: x-axis?

File: packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ * between 10 and 20, and in red where its y-coordinates are between 20 and 40.
+ * If the timeseries has values outside these intervals, those line segments
+ * won't be rendered.
+ *
+ * @param timeseries Timeseries to represent as a line.
+ * @param xScale d3-scale to transform point dates to pixel positions on the x-axis.
+ * @param yScale d3-scale to transform point values to pixel positions on the y-axis.
+ * @param intervals intervals that define the color of the line for different segments.
+ *
+ * @returns An SVG group containing the line.
```
Comment (mikelehen) [inline packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx]: Note that nothing in the description mentions "thresholds" which suggests we might want to rename this component.  LineIntervalChart or LineColoredIntervalsChart or something?  Not sure.

**Review by mikelehen — COMMENTED** [review]
Couple doc nits, but otherwise looks good!

File: packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx (hunk 4 above)
Comment (pnavarrc) [reply packages/ui-components/src/components/LineThresholdChart/LineThresholdChart.tsx]: Good call! That's probably better, renaming!

**Pull request merged**